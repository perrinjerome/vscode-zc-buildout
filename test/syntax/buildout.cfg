# comment
; comment
[section]
option = value
# comment
; comment
option =
  # not comment (after space)
	# not comment (aftar tabs)

[substitutions]
option = va_ue # not-comment
option = ${section:option}
option = text ${section:option}
option = ${section:option} text
option = text ${section:option} ${section:option}
option = ${section:option} text ${section:option}
option = ${section:option} ${section:option} text
option = text ${section:option} text ${section:option}
option = text ${section:option} text ${section:option} text
option = ${section:option} text ${section:option} text


[conditional-section:condition]


[strings]
line = "single line"
without-end = "without end
this = is not string
