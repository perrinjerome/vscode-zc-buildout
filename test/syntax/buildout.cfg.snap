># comment
#^ source.zc-buildout-jinja comment.line.number-sign.zc-buildout punctuation.definition.comment.zc-buildout
# ^^^^^^^^ source.zc-buildout-jinja comment.line.number-sign.zc-buildout
>[section]
#^ source.zc-buildout-jinja entity.name.type.section-name.zc-buildout punctuation.definition.entity.zc-buildout
# ^^^^^^^ source.zc-buildout-jinja entity.name.type.section-name.zc-buildout entity.name.type.section-name.zc-buildout
#        ^ source.zc-buildout-jinja entity.name.type.section-name.zc-buildout punctuation.definition.entity.zc-buildout
>option = value
#^^^^^^ source.zc-buildout-jinja keyword.other.definition.zc-buildout
#      ^ source.zc-buildout-jinja
#       ^ source.zc-buildout-jinja punctuation.separator.key-value.zc-buildout
#        ^^^^^^^ source.zc-buildout-jinja
># comment
#^ source.zc-buildout-jinja comment.line.number-sign.zc-buildout punctuation.definition.comment.zc-buildout
# ^^^^^^^^ source.zc-buildout-jinja comment.line.number-sign.zc-buildout
>
>[substitutions]
#^ source.zc-buildout-jinja entity.name.type.section-name.zc-buildout punctuation.definition.entity.zc-buildout
# ^^^^^^^^^^^^^ source.zc-buildout-jinja entity.name.type.section-name.zc-buildout entity.name.type.section-name.zc-buildout
#              ^ source.zc-buildout-jinja entity.name.type.section-name.zc-buildout punctuation.definition.entity.zc-buildout
>option = va_ue # not-comment
#^^^^^^ source.zc-buildout-jinja keyword.other.definition.zc-buildout
#      ^ source.zc-buildout-jinja
#       ^ source.zc-buildout-jinja punctuation.separator.key-value.zc-buildout
#        ^^^^^^^^^^^^^^^^^^^^^ source.zc-buildout-jinja
>option = ${section:option}
#^^^^^^ source.zc-buildout-jinja keyword.other.definition.zc-buildout
#      ^ source.zc-buildout-jinja
#       ^ source.zc-buildout-jinja punctuation.separator.key-value.zc-buildout
#        ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#           ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                         ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
>option = text ${section:option}
#^^^^^^ source.zc-buildout-jinja keyword.other.definition.zc-buildout
#      ^ source.zc-buildout-jinja
#       ^ source.zc-buildout-jinja punctuation.separator.key-value.zc-buildout
#        ^^^^^ source.zc-buildout-jinja
#             ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#                ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                              ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
>option = ${section:option} text
#^^^^^^ source.zc-buildout-jinja keyword.other.definition.zc-buildout
#      ^ source.zc-buildout-jinja
#       ^ source.zc-buildout-jinja punctuation.separator.key-value.zc-buildout
#        ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#           ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                         ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
#                          ^^^^^^ source.zc-buildout-jinja
>option = text ${section:option} ${section:option}
#^^^^^^ source.zc-buildout-jinja keyword.other.definition.zc-buildout
#      ^ source.zc-buildout-jinja
#       ^ source.zc-buildout-jinja punctuation.separator.key-value.zc-buildout
#        ^^^^^ source.zc-buildout-jinja
#             ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#                ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                              ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
#                               ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#                                  ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                                                ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
>option = ${section:option} text ${section:option}
#^^^^^^ source.zc-buildout-jinja keyword.other.definition.zc-buildout
#      ^ source.zc-buildout-jinja
#       ^ source.zc-buildout-jinja punctuation.separator.key-value.zc-buildout
#        ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#           ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                         ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
#                          ^^^^^ source.zc-buildout-jinja
#                               ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#                                  ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                                                ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
>option = ${section:option} ${section:option} text
#^^^^^^ source.zc-buildout-jinja keyword.other.definition.zc-buildout
#      ^ source.zc-buildout-jinja
#       ^ source.zc-buildout-jinja punctuation.separator.key-value.zc-buildout
#        ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#           ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                         ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
#                          ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#                             ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                                           ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
#                                            ^^^^^^ source.zc-buildout-jinja
>option = text ${section:option} text ${section:option}
#^^^^^^ source.zc-buildout-jinja keyword.other.definition.zc-buildout
#      ^ source.zc-buildout-jinja
#       ^ source.zc-buildout-jinja punctuation.separator.key-value.zc-buildout
#        ^^^^^ source.zc-buildout-jinja
#             ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#                ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                              ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
#                               ^^^^^ source.zc-buildout-jinja
#                                    ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#                                       ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                                                     ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
>option = text ${section:option} text ${section:option} text
#^^^^^^ source.zc-buildout-jinja keyword.other.definition.zc-buildout
#      ^ source.zc-buildout-jinja
#       ^ source.zc-buildout-jinja punctuation.separator.key-value.zc-buildout
#        ^^^^^ source.zc-buildout-jinja
#             ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#                ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                              ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
#                               ^^^^^ source.zc-buildout-jinja
#                                    ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#                                       ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                                                     ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
#                                                      ^^^^^^ source.zc-buildout-jinja
>option = ${section:option} text ${section:option} text
#^^^^^^ source.zc-buildout-jinja keyword.other.definition.zc-buildout
#      ^ source.zc-buildout-jinja
#       ^ source.zc-buildout-jinja punctuation.separator.key-value.zc-buildout
#        ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#           ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                         ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
#                          ^^^^^ source.zc-buildout-jinja
#                               ^^^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.begin.zc-buildout
#                                  ^^^^^^^^^^^^^^ source.zc-buildout-jinja variable.other.single.zc-buildout
#                                                ^ source.zc-buildout-jinja variable.other.single.zc-buildout punctuation.definition.string.end.zc-buildout
#                                                 ^^^^^^ source.zc-buildout-jinja
>
>
>[conditional-section:condition]
#^ source.zc-buildout-jinja entity.name.type.section-name.zc-buildout punctuation.definition.entity.zc-buildout
# ^^^^^^^^^^^^^^^^^^^ source.zc-buildout-jinja entity.name.type.section-name.zc-buildout entity.name.type.section-name.zc-buildout
#                    ^^^^^^^^^^ source.zc-buildout-jinja entity.name.type.section-name.zc-buildout variable.parameter.section-condition.zc-buildout
#                              ^ source.zc-buildout-jinja entity.name.type.section-name.zc-buildout punctuation.definition.entity.zc-buildout
>
>